src/Core__List.res

```diff
@@ -94,7 +94,7 @@ let head = x =>
 
 let headExn = x =>
   switch x {
-  | list{} => raise(Not_found)
+  | list{} => Core__Error.panic("List.headExn: list is empty")
```

Comment:
With changes in the `List` module for example, I can no longer do pattern matching?

Reply:
Pattern matching doesn't use any library-defined functions, so this shouldn't affect that. If I understood your question correctly.

Reply:
Yeah, that would no longer work. You should still be able to match on `Exn.Error` though, but it will be less precise:

```rescript
let result =
  try {
    List.headExn(list{1, 2})
  } catch {
  | Exn.Error(_) => 0 // Default value if headExn throws
  }
```

But what you should usually do instead in this case is (both before and after this change) is use `List.head`:

```rescript
let result =
  switch List.head(list{1, 2}) {
  | Some(n) => n
  | None=> 0
  }
```

Or better yet:

```rescript
let result = List.head(list{1, 2})->Option.getWithDefault(0) 
```